src/SWP/Bundle/TemplateEngineBundle/Controller/MenuNodeController.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Template Engine Bundle.
+ *
+ * Copyright 2015 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\TemplateEngineBundle\Controller;
+
+use Doctrine\ODM\PHPCR\DocumentManager;
+use FOS\RestBundle\Controller\FOSRestController;
+use FOS\RestBundle\View\View;
+use Nelmio\ApiDocBundle\Annotation\ApiDoc;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Cmf\Bundle\MenuBundle\Doctrine\Phpcr\MenuNode;
+use Symfony\Component\HttpFoundation\Request;
+use SWP\Bundle\TemplateEngineBundle\Form\Type\MenuNodeType;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\HttpKernel\Exception\UnprocessableEntityHttpException;
+
+class MenuNodeController extends FOSRestController
+{
+    /**
+     * Lists all registered menus.
+     *
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Lists all registered menu nodes of a given menu",
+     *     statusCodes={
+     *         200="Returned on success.",
+     *         404="No menu nodes found."
+     *     }
+     * )
+     * @Route("/api/{version}/templates/menunodes/{menuId}", options={"expose"=true}, defaults={"version"="v1","menuId"=null}, name="swp_api_templates_list_menu_nodes")
+     * @Method("GET")
+     */
+    public function listAction(Request $request, $menuId)
+    {
+        if (!$menuId) {
+            throw new UnprocessableEntityHttpException('You need to provide menu name (name).');
+        }
+
+        $mp = $this->get('swp_template_engine.menu_provider');
```

Comment:
this variable is two short imho, min 3 chars is advised, what about naming it `$menuProvider`?